### PR TITLE
migrations + models

### PR DESCRIPTION
**Pull request by SotirisTavianatos**

### CODE REVIEW

File: migrations/2022_05_08_031822_create_stats_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('stats', function (Blueprint $table) {
+            $table->id();
+            $table->integer('points');
+            $table->integer('player_id')->unsigned();
```
Comment (SotirisBK) [inline migrations/2022_05_08_031822_create_stats_table.php]: you need to set this foreign key to the correct table

Comment (SotirisBK) [reply migrations/2022_05_08_031822_create_stats_table.php]: For this to work I guess you will need the players table to be created first I guess

File: migrations/2022_05_09_233348_create_comments_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('comments', function (Blueprint $table) {
+            $table->id();
+            $table->string('content');
+            $table->integer('player_id');
+            $table->integer('user_id');
```
Comment (SotirisBK) [inline migrations/2022_05_09_233348_create_comments_table.php]: set relations for those
```
$table->index("user_id");
            $table->foreign("user_id")
                ->references('id')->on('users')
                ->onDelete('cascade');
```

The above sample first creates the index on the field and then adds the foreign key relation. If also any other field requires index eg for faster search make sure you add this as well

File: migrations/2022_05_14_181416_create_comment_user_table.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('comment_user', function (Blueprint $table) {
+            $table->id();
+            $table->integer('comment_id');
+            $table->integer('user_id');
```
Comment (SotirisBK) [inline migrations/2022_05_14_181416_create_comment_user_table.php]: relations

File: migrations/2022_05_14_181534_create_player_team_table.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('player_team', function (Blueprint $table) {
+            $table->id();
+            $table->integer('player_id');
+            $table->integer('team_id');
```
Comment (SotirisBK) [inline migrations/2022_05_14_181534_create_player_team_table.php]: relations

File: Models/Player.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    public function comments(){
+        return $this->hasMany('App\Models\Comment','player_id');    
+    }
+    //player's teams
+    public function teams(){
+        return $this->belongsToMany('App\Models\Team');    
+    }
+    //player's current team
+    public function currentteam(){
+        return DB::table('player_team')->join('teams','team_id','=','player_team.team_id')->whereNull('until')->value('name');
```
Comment (SotirisBK) [inline Models/Player.php]: a good idea would be to also have not null the from date n the above query

File: migrations/2022_05_14_180552_create_notifications_table.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('notifications', function (Blueprint $table) {
+            $table->id();
```
Comment (SotirisBK) [inline migrations/2022_05_14_180552_create_notifications_table.php:17]: the notification should have:
- text
- user_id (and a db relation)
- status( if your db supports enum use it) which could be NEW:0, SENT:1, READ:2 or you could even use separate fields per each state(boolean)

File: migrations/2022_05_09_233348_create_comments_table.php (hunk 2 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_09_233348_create_comments_table.php]: done

File: migrations/2022_05_14_181416_create_comment_user_table.php (hunk 3 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_14_181416_create_comment_user_table.php]: done

File: migrations/2022_05_14_181534_create_player_team_table.php (hunk 4 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_14_181534_create_player_team_table.php]: done

File: Models/Player.php (hunk 5 above)
Comment (SotirisTavianatos) [reply Models/Player.php]: done

File: migrations/2022_05_14_180552_create_notifications_table.php (hunk 6 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_14_180552_create_notifications_table.php:17]: done

File: migrations/2022_05_14_181416_create_comment_user_table.php [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('comment_user', function (Blueprint $table) {
```
Comment (sotirisboukouvalas) [inline migrations/2022_05_14_181416_create_comment_user_table.php]: One last think I now understood is that you are using this m2m table to have the relation for likes? the name is not ideal although aligns with laravel standards. we need though to take control of this and have a name that is better fit. Maybe something like `comment_user_likes`

File: Models/Notification.php [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+use Illuminate\Database\Eloquent\Model;
+
+class Notification extends Model
+{
+    use HasFactory;
+    //about what comment is this notification
+    public function comment()
+    {
+        return $this->belongsTo('App\Models\Comment','comment_id');    
+    }
```
Comment (sotirisboukouvalas) [inline Models/Notification.php:20]: Where is the relation with user?

File: migrations/2022_05_14_180552_create_notifications_table.php (hunk 6 above)
Comment (sotirisboukouvalas) [reply migrations/2022_05_14_180552_create_notifications_table.php:17]: you need first to declare the user_id
then to index it and then to add the foreign key eg.
```
Schema::create('pets', function (Blueprint $table) {
            $table->bigIncrements('id');
            $table->string('name', 40)->nullable(true);
            $table->string('type',10)->nullable(false);
            $table->string('breed', 100)->nullable(false);
            $table->string('chip', 100)->nullable(true);
            $table->date('birthday')->nullable(true);
            $table->smallInteger('weight')->nullable(true);
            $table->unsignedBigInteger("client_id")->nullable(false);
            $table->boolean('alive')->default(true);
            $table->timestamps();
            $table->softDeletes();

            $table->index("client_id");
            $table->foreign("client_id")
                ->references('id')->on('clients')
                ->onDelete('cascade');
        });
```

Comment (sotirisboukouvalas) [reply migrations/2022_05_14_180552_create_notifications_table.php:17]: we do the same for all columns that are relations. so for comment id as well

File: migrations/2022_05_08_031822_create_stats_table.php (hunk 1 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_08_031822_create_stats_table.php]: done

File: migrations/2022_05_14_180552_create_notifications_table.php (hunk 6 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_14_180552_create_notifications_table.php:17]: done

File: migrations/2022_05_14_181416_create_comment_user_table.php (hunk 7 above)
Comment (SotirisTavianatos) [reply migrations/2022_05_14_181416_create_comment_user_table.php]: ok renamed it as comment_user_likes

File: Models/Notification.php (hunk 8 above)
Comment (SotirisTavianatos) [reply Models/Notification.php:20]: relation has been added

**Pull request merged**